### PR TITLE
Config migrate

### PR DESCRIPTION
**Pull request by arturrez**
```
❯ ./bin/avalanche config migrate
-----------------------------------------------------------------------
WARNING: Depricated configuration file was found in /Users/artur/.avalanche-cli
Please run avalanche config migrate to migrate it to new default location config
-----------------------------------------------------------------------
Configuration migrated to /Users/artur/.avalanche-cli/config
Depricated configuration file /Users/artur/.avalanche-cli.json removed
```

### CODE REVIEW

File: cmd/configcmd/migrate.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		RunE:         migrateConfig,
+		SilenceUsage: true,
+	}
+	return cmd
+}
+
+func migrateConfig(_ *cobra.Command, _ []string) error {
+	home, err := os.UserHomeDir()
+	cobra.CheckErr(err)
+	oldConfigFilename := fmt.Sprintf("%s/%s.%s", home, constants.OldConfigFileName, constants.DefaultConfigFileType)
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/configcmd/migrate.go:32]: we can have constants for all this absolute paths

File: cmd/configcmd/migrate.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	if !app.ConfigFileExists(oldConfigFilename) {
+		return fmt.Errorf("depricated configuration file %s does not exist", oldConfigFilename)
+	} else {
+		// load old config
+		viper.SetConfigFile(oldConfigFilename)
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/configcmd/migrate.go:37]: I would avoid viper and keep it encaplusated on app

File: pkg/application/app.go [hunk 3]
```diff
@@ -602,20 +602,11 @@ func (*Avalanche) writeFile(path string, bytes []byte) error {
 	return os.WriteFile(path, bytes, constants.WriteReadReadPerms)
 }
 
-func (app *Avalanche) LoadConfig() (models.Config, error) {
+func (app *Avalanche) ConfigFileExists(path string) bool {
```
Comment (felipemadero) [inline pkg/application/app.go:605]: avoid passing in a path. check config exists from a new app variable that refers to the config file path, to be updated with app.SetConfigPath

File: pkg/application/app.go [hunk 4]
```diff
@@ -625,8 +616,11 @@ func (app *Avalanche) ConfigFileExists() bool {
 	return true
 }
 
-func (app *Avalanche) WriteConfigFile(bytes []byte) error {
+func (app *Avalanche) WriteConfigFile(bytes []byte, path string) error {
```
Comment (felipemadero) [inline pkg/application/app.go:619]: avoid passing in a path

File: pkg/application/app.go [hunk 5]
```diff
@@ -625,8 +616,11 @@ func (app *Avalanche) ConfigFileExists() bool {
...
 	configPath := app.GetConfigPath()
+	if path != "" {
+		configPath = path
+	}
 	return app.writeFile(configPath, bytes)
 }
```
Comment (felipemadero) [inline pkg/application/app.go:625]: add a function app.LoadConfig and app.SetConfigValue, probably another app.GetConfigValue

Comment (felipemadero) [reply pkg/application/app.go:625]: I believe we can also have a new Config struct for this, encapsulating viper, with an instance of it on app

File: pkg/constants/constants.go [hunk 6]
```diff
@@ -133,7 +133,9 @@ const (
 	AnsibleSSHInventoryParams                    = "-o StrictHostKeyChecking=no"
 	AnsibleExtraVarsFlag                         = "--extra-vars"
 	CloudCLIBranch                               = "custom-vms-cloud-10"
-	DefaultConfigFileName                        = ".avalanche-cli"
+	MetricsEnabled                               = "MetricsEnabled"
```
Comment (felipemadero) [inline pkg/constants/constants.go:136]: add constants for old default configs paths, and new default config path, absolute ones

Comment (felipemadero) [inline pkg/constants/constants.go:136]: add some Config ref to MetricsEnabled name

File: pkg/utils/metrics.go [hunk 7]
```diff
@@ -55,10 +55,11 @@ func PrintMetricsOptOutPrompt() {
 		"You can also read our privacy statement <https://www.avalabs.org/privacy-policy> to learn more.\n")
 }
 
-func saveMetricsConfig(app *application.Avalanche, metricsEnabled bool) {
-	config := models.Config{MetricsEnabled: metricsEnabled}
-	jsonBytes, _ := json.Marshal(&config)
-	_ = app.WriteConfigFile(jsonBytes)
+func saveMetricsConfig(metricsEnabled bool) {
+	viper.Set(constants.MetricsEnabled, metricsEnabled)
```
Comment (felipemadero) [inline pkg/utils/metrics.go:59]: app.SetConfigValue

File: pkg/utils/metrics.go [hunk 8]
```diff
@@ -73,21 +74,16 @@ func HandleUserMetricsPreference(app *application.Avalanche) error {
...
 
-func userIsOptedIn(app *application.Avalanche) bool {
-	// if config file is not found or unable to be read, will return false (user is not opted in)
-	config, err := app.LoadConfig()
-	if err != nil {
-		return false
-	}
-	return config.MetricsEnabled
+func userIsOptedIn() bool {
+	return viper.GetBool(constants.MetricsEnabled)
```
Comment (felipemadero) [inline pkg/utils/metrics.go:82]: call a function on app, maybe app.GetConfigValue, that returns an interface to be converted afterwards

File: cmd/configcmd/migrate.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/configcmd/migrate.go:37]: viper makes things much cleaner. no point of not using it if we have already

File: cmd/root.go [hunk 9]
```diff
@@ -309,9 +309,17 @@ func initConfig() {
 		// Search for default config.
 		home, err := os.UserHomeDir()
 		cobra.CheckErr(err)
-		viper.AddConfigPath(home)
-		viper.SetConfigType(constants.DefaultConfigFileType)
+		viper.AddConfigPath(fmt.Sprintf("%s/%s", home, constants.BaseDirName))
 		viper.SetConfigName(constants.DefaultConfigFileName)
+		viper.SetConfigType(constants.DefaultConfigFileType)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/root.go:314]: i think there is a easier fix

File: pkg/utils/metrics.go (hunk 8 above)
Comment (arturrez) [reply pkg/utils/metrics.go:82]: not sure how you want to implement this

**Review by felipemadero — APPROVED** [review]

**Pull request merged**